### PR TITLE
improve tslint rules / use latest ruleset

### PR DESCRIPTION
**Pull request by lukaselmer**

### CODE REVIEW

File: src/app/controllers/edit_controller.spec.ts [hunk 1]
```diff
@@ -44,8 +44,9 @@ describe('EditController', function () {
     expect(uploadSpy.calls.count()).toBe(1);
   });
 
-  it('loads the upload', function () {
-  });
+  // cannot be tested?
```
Comment (sihu) [inline src/app/controllers/edit_controller.spec.ts]: should that be part of the dropzone inclusion?

File: src/app/controllers/edit_controller.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ class EditController {
     this.loadDependencies();
 
     this.loadingCallback.done(() =>
-      this.preparer.prepare(dom, (dom:DomContentBlock, newContent:string) => this.editContent(dom, newContent)));
+      this.preparer.prepare(dom, (domB:DomContentBlock, newContent:string) => this.editContent(domB, newContent)));
```
Comment (sihu) [inline src/app/controllers/edit_controller.ts]: not a real understandable variable name at the first moment

**Comment by sihu** [comment]
LGTM!  👍  there is only the mentioned points i would like to discuss before merging! But this PR brings a lot of improvements! thx!

File: src/app/controllers/edit_controller.spec.ts (hunk 1 above)
Comment (lukaselmer) [inline src/app/controllers/edit_controller.spec.ts]: i don't know. i just saw the method there and it was not doing anything?

File: src/app/controllers/edit_controller.ts (hunk 2 above)
Comment (lukaselmer) [inline src/app/controllers/edit_controller.ts]: agree, fixed!

File: src/app/controllers/edit_controller.spec.ts (hunk 1 above)
Comment (lukaselmer) [inline src/app/controllers/edit_controller.spec.ts]: but let's remove it - it seems that it's unused!

**Pull request merged**